src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1269,26 +1277,115 @@ public static bool IsPatternContext(this SyntaxTree syntaxTree, SyntaxToken left
 
             // e switch { $$
             // e switch { ..., $$
-            if (leftToken.IsKind(SyntaxKind.OpenBraceToken, SyntaxKind.CommaToken) && leftToken.Parent.IsKind(SyntaxKind.SwitchExpression))
+            if (leftToken.IsKind(SyntaxKind.OpenBraceToken, SyntaxKind.CommaToken) && node.IsKind(SyntaxKind.SwitchExpression))
             {
                 return true;
             }
 
             // e is ($$
             // e is (..., $$
-            if (leftToken.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken) && leftToken.Parent.IsKind(SyntaxKind.PositionalPatternClause))
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken) && node.IsKind(SyntaxKind.PositionalPatternClause))
             {
                 return true;
             }
 
             // e is { P: $$
             // e is { ..., P: $$
-            if (leftToken.IsKind(SyntaxKind.ColonToken) && leftToken.Parent.IsKind(SyntaxKind.NameColon) &&
-                leftToken.Parent.IsParentKind(SyntaxKind.Subpattern))
+            if (leftToken.IsKind(SyntaxKind.ColonToken) && node.IsKind(SyntaxKind.NameColon) &&
+                node.IsParentKind(SyntaxKind.Subpattern))
+            {
+                return true;
+            }
+
+#if !CODE_STYLE
+            // e switch { ($$
+            // e switch { ..., ($$
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken) && node.IsParentKind(SyntaxKind.SwitchExpressionArm))
+            {
+                return true;
+            }
+
+            // case ($
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken) && node.IsParentKind(SyntaxKind.SwitchSection))
+            {
+                return true;
+            }
+
+            // e is ((($$
+            // e is { P: ($$
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken) && node.IsKind(SyntaxKind.ConstantPattern))
             {
                 return true;
             }
 
+            // if (e is { P: (1 or $$
+            if ((leftToken.IsKind(SyntaxKind.AndKeyword) || leftToken.IsKind(SyntaxKind.OrKeyword)) &&
+                node.IsParentKind(SyntaxKind.ParenthesizedPattern))
```

Comment:
i don't think you need `node.IsParentKind(SyntaxKind.ParenthesizedPattern)`.  instead you likely want `leftToken.Parent is BinaryPattern`.